eipw-lint/src/lints/markdown/no_emphasis_keywords.rs

```diff
@@ -0,0 +1,124 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use eipw_snippets::Level;
+
+use comrak::nodes::Ast;
+
+use crate::lints::{Context, Error, Lint};
+use crate::tree::{self, Next, TraverseExt};
+
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+use std::fmt::Debug;
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+#[cfg_attr(feature = "schema-version", derive(schemars::JsonSchema))]
+pub struct NoEmphasisKeywords;
+
+impl Lint for NoEmphasisKeywords {
+    fn lint<'a>(&self, slug: &'a str, ctx: &Context<'a, '_>) -> Result<(), Error> {
+        let mut visitor = Visitor {
+            ctx,
+            slug,
+            in_emphasis: false,
+            emphasis_type: EmphasisType::None,
+            uppercase_regex: Regex::new(r"\b[A-Z]{2,}\b").map_err(Error::custom)?,
+        };
+        ctx.body().traverse().visit(&mut visitor)?;
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+enum EmphasisType {
+    None,
+    Bold,
+    Italic,
+}
+
+impl EmphasisType {
+    fn description(&self) -> &'static str {
+        match self {
+            EmphasisType::None => "",
+            EmphasisType::Bold => "bold",
+            EmphasisType::Italic => "italic",
+        }
+    }
+}
+
+struct Visitor<'a, 'b, 'c> {
+    ctx: &'c Context<'a, 'b>,
+    slug: &'c str,
+    in_emphasis: bool,
+    emphasis_type: EmphasisType,
```

Comment:
```suggestion
    emphasis_type: Option<EmphasisType>,
```

It's less error prone to combine the bool with the kind. This way you can't accidentally represent the state `in_emphasis == false && emphasis_type != None`.